numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -3628,11 +3672,19 @@ PyArray_GetDTypeTransferFunction(int aligned,
             }
         }
 
-        /* The special types, which have no byte-order */
+        /* The special types, which have no or subelement byte-order */
         switch (src_type_num) {
             case NPY_VOID:
```

Comment:
while `NPY_VOID` will always fail the if-statement if the next `NPY_UNICODE` case (there's no bug), maybe it is clearer to put the `NPY_VOID` case below the unicode case?